### PR TITLE
rename "sector store type" to "proofs mode"

### PR DESCRIPTION
**Pull request by laser**
Makes incremental progress towards #2530.

## What's in this PR?

- rename "SectorStoreType" to "ProofsMode"

## Why is this PR needed?

- the old name didn't make sense

### CODE REVIEW

**Comment by laser** [comment]
cc @rkowalick

File: proofs/interface.go [hunk 1]
```diff
@@ -36,7 +36,9 @@ type Verifier interface {
 	VerifySeal(VerifySealRequest) (VerifySealResponse, error)
 }
 
-// Mode configures the behavior of the SectorStore used by the SectorBuilder.
+// Mode configures sealing, sector packing, PoSt generation and other behaviors
```
Comment (ZenGround0) [inline proofs/interface.go]: Great comment

**Comment by rkowalick** [comment]
As per the quote below, can you rename `proofs.Test` and `proofs.Live`?
> Could I convince you to use the term "proofs mode" and type `proofs.Mode` instead of "sector store type" (`proofs.SectorStoreType`) in your PR?
>
> The two members of that type would be `LiveMode` and `TestMode`.

File: proofs/interface.go (hunk 1 above)
Comment (laser) [reply proofs/interface.go]: thnx sir

**Comment by laser** [comment]
@rkowalick - 

> As per the quote below, can you rename proofs.Test and proofs.Live?

done!

**Pull request merged**